pkg/agreementreceiver/server.go

```diff
@@ -51,16 +85,17 @@ func (s *Server) BandwidthAgreements(stream pb.Bandwidth_BandwidthAgreementsServ
 		case err := <-errch:
 			return err
 		case <-ctx.Done():
+			s.logger.Debug("ctx<-Done()")
 			return nil
 		case agreement := <-ch:
 			go func() {
-				fmt.Println(agreement)
-				//TODO: write to DB
-				//err = s.DB.WriteAgreement(agreement)
-				// if err != nil {
-				// 	return err
-				// }
+				s.logger.Debug("about to create a postgres entry")
+				_, err = s.Create(ctx, agreement)
+				if err != nil {
+					errch <- err
```

Comment:
Looks like the entire process ends when errch gets an error. Should We just log errors instead? @egonelbre

Reply:
added logger.Error messages at places where err!=nil

Reply:
I was only reading a small part of the function. These errors passed into the errch are fine!